modules/RTC/TrackStreamingStatus.js

```diff
@@ -0,0 +1,738 @@
+         * <tt>Date.now()</tt>.
+         *
+         * FIXME merge this logic with NO_DATA_FROM_SOURCE event
+         *       implemented in JitsiLocalTrack by extending the event to
+         *       the remote track and allowing to set different timeout for
+         *       local and remote tracks.
+         *
+         * @type {number}
+         */
+        this.rtcMutedTimestamp = null;
+        logger.info(`RtcMuteTimeout set to: ${this.rtcMuteTimeout}`);
+
+        /**
+         * This holds the timestamp indicating when remote video track
+         * entered forwarded sources set. Track entering forwardedSources
+         *  will have connection status restoring and when we start receiving
+         * video will become active, but if video is not received for certain
+         * time {@link DEFAULT_RESTORING_TIMEOUT} that track connection status
+         * will become interrupted.
+         *
+         * @type {number}
+         */
+        this.enteredForwardedSourcesTimestamp = null;
+
+        /**
+         * This holds the restoring timeout callback ID
+         * scheduled using window.setTimeout.
+         *
+         * @type {number}
+         */
+        this.restoringTimer = null;
+
+        /**
+         * This holds the current connection status (along with all the internal events that happen
+         * while in that state).
+         *
+         * The goal is to send this information to the analytics backend for post-mortem analysis.
+         */
+        this.connectionStatusMap = {};
+    }
+
+    /**
+     * Gets the video frozen timeout for given source name.
+     * @return {number} how long are we going to wait since RTC video muted
+     * even, before a video track is considered frozen.
+     * @private
+     */
+    _getVideoFrozenTimeout() {
+        const sourceName = this.track.getSourceName();
+
+        return this.rtc.isInForwardedSources(sourceName)
+            ? this.rtcMuteTimeout
+            : this.conference.isP2PActive() ? this.p2pRtcMuteTimeout : this.outOfLastNTimeout;
+    }
+
+    /**
+     * Initializes <tt>TrackStreamingStatus</tt> and bind required event
+     * listeners.
+     */
+    init() {
+        // Handles P2P status changes
+        this._onP2PStatus = this.figureOutConnectionStatus.bind(this);
+        this.conference.on(JitsiConferenceEvents.P2P_STATUS, this._onP2PStatus);
+
+        // Used to send analytics events for the participant that left the call.
+        this._onUserLeft = this.onUserLeft.bind(this);
+        this.conference.on(JitsiConferenceEvents.USER_LEFT, this._onUserLeft);
+
+        // On some browsers MediaStreamTrack trigger "onmute"/"onunmute"
+        // events for video type tracks when they stop receiving data which is
+        // often a sign that remote user is having connectivity issues
+        if (browser.supportsVideoMuteOnConnInterrupted()) {
+
+            this._onTrackRtcMuted = this.onTrackRtcMuted.bind(this);
+            this.rtc.addListener(
```

Comment:
Will it make more sense to move all this logic to register rtc event listeners (`REMOTE_TRACK_MUTE, REMOTE_TRACK_UNMUTE, LASTN_VALUE_CHANGED`) into `JitsiConference`? Maybe in `_registerRtcListeners` since it is already behind the feature flag?

Reply:
I think it's okay to register here. Otherwise the `JitsiConference` would have to call into this class which would tie them together (right now `JitsiConference` is not aware of this class - I think). Why do you think it's better to do it there? Is it because the events appear to be global in a way?

Reply:
Yes in a way it does feel like these are global events so I was hesitant on having them in here but I see both of your points now